File: 161. One Edit Distance/isOneEditDistance.py
class Solution:
  def isOneEditDistance(self, s: str, t: str) -> bool:
    slen, tlen = len(s), len(t)
    
    if slen > tlen:
      return self.isOneEditDistance(t, s)
    
    if tlen - slen > 1:
      return False
    
    for i in range(slen):
      if s[i] != t[i]:
        if slen == tlen:
          return s[i + 1:] == t[i + 1:]
        else:
          return s[i:] == t[i + 1:]
    
    return slen + 1 == tlen
    
s = "ab"
t = "acb"
solution = Solution()
print('isOneEditDistance: ', solution.isOneEditDistance(s, t))

File: 161. One Edit Distance/isOneEditDistance.js
/**
 * @param {string} s
 * @param {string} t
 * @return {boolean}
 */
const isOneEditDistance = (s, t) => {
  const sLen = s.length;
  const tLen = t.length;

  if (sLen > tLen) {
    return isOneEditDistance(t, s);
  }

  if (tLen - sLen > 1) {
    return false;
  }

  for (let i = 0; i < sLen; i++) {
    if (s[i] !== t[i]) {
      if (sLen === tLen) {
        return s.slice(i + 1) === t.slice(i + 1);
      } else {
        return s.slice(i) === t.slice(i + 1);
      }
    }
  }

  return sLen + 1 === tLen;
};

const s = "ab";
const t = "acb";
console.log(isOneEditDistance(s, t)); // true
